lib/trino-hdfs/src/main/java/io/trino/hdfs/TrinoFileSystemCache.java

```diff
@@ -429,6 +429,34 @@ public InputStream getWrappedStream()
         }
     }
 
+    private static class RemoteIteratorWrapper
+            implements RemoteIterator<LocatedFileStatus>
+    {
+        private final RemoteIterator<LocatedFileStatus> delegate;
+        @SuppressWarnings({"FieldCanBeLocal", "unused"})
+        private final FileSystem fileSystem;
```

Comment:
This should have a comment explaining why we need this class and field. Otherwise, someone may come around in the future and clean up this unused field.

Reply:
agreed. i think it went this way for consistency https://github.com/trinodb/trino/blob/b25d35939d2e7236cbac51bcd72aa7a40d9031f7/lib/trino-hdfs/src/main/java/io/trino/hdfs/TrinoFileSystemCache.java#L395-L431

Reply:
https://github.com/trinodb/trino/pull/17629